packages/amplify-provider-awscloudformation/src/pre-push-cfn-processor/cfn-pre-processor.ts

```diff
@@ -24,3 +37,140 @@ export async function preProcessCFNTemplate(filePath: string): Promise<string> {
   await writeCFNTemplate(cfnTemplate, newPath, { templateFormat });
   return newPath;
 }
+
+//get data from custom polcies file and write custom policies to CFN template
+export async function writeCustomPoliciesToCFNTemplate(
+  resourceName: string,
+  service: string,
+  resourceDir: string,
+  cfnFile: string,
+  category: string,
+  filePath: string
+) {
+  const { templateFormat, cfnTemplate } = await readCFNTemplate(path.join(resourceDir, cfnFile));
+  const customPolicies = stateManager.getCustomPolicies(service, category, resourceName);
+  if (!customPolicies || !this.validateCustomPoliciesSchema(customPolicies)) {
+    return;
```

Comment:
if the customPolicies do not pass validation, I think we should error with a message saying what was invalid rather than silently continue

Reply:
Customers might not specify anything in the custom policies. This is to validate if there is anything we should merge to the CFN template, also if the schema of the policies is correct. So I just throw a warning message for it probably nothing needed to be merged. Do you think we should separate this?